src/Raven.Client/Documents/TcpUtils.cs

```diff
@@ -62,7 +61,19 @@ internal static void SetTimeouts(TcpClient client, TimeSpan timeout)
                 throw;
             }
         }
-        
+
+        public static Task ConnectAsync(TcpClient tcpClient, string url)
+        {
+            var uri = new Uri(url);
+            if (uri.Host[0] == '[') // IPv6 format, as [::1]
```

Comment:
IIRC in Uri we have a HostNameType (with Ipv6 as a one of the enumeration options). Maybe it can be used?

Reply:
I didn't knew it exists, thanks! I'll use it.